packages/jest-snapshot/src/InlineSnapshots.ts

```diff
@@ -6,109 +6,130 @@
+const generate: typeof import('@babel/generator')['default'] = require(require.resolve(
+  '@babel/generator',
+  {[Symbol.for('OUTSIDE_JEST_VM_RESOLVE_OPTION')]: true} as any,
+)).default;
+const {file, templateElement, templateLiteral} = require(require.resolve(
+  '@babel/types',
+  {
+    [Symbol.for('OUTSIDE_JEST_VM_RESOLVE_OPTION')]: true,
+  } as any,
+));
+const {parseSync} = require(require.resolve('@babel/core', {
+  [Symbol.for('OUTSIDE_JEST_VM_RESOLVE_OPTION')]: true,
+} as any)) as typeof import('@babel/core');
+
 export type InlineSnapshot = {
   snapshot: string;
   frame: Frame;
+  node?: Expression;
 };
 
 export function saveInlineSnapshots(
   snapshots: Array<InlineSnapshot>,
-  prettier: Prettier | null,
-  babelTraverse: BabelTraverse,
+  prettierPath: Config.Path,
 ): void {
-  if (!prettier) {
-    throw new Error(
-      `Jest: Inline Snapshots requires Prettier.\n` +
-        `Please ensure "prettier" is installed in your project.`,
-    );
-  }
-
-  // Custom parser API was added in 1.5.0
-  if (semver.lt(prettier.version, '1.5.0')) {
-    throw new Error(
-      `Jest: Inline Snapshots require prettier>=1.5.0.\n` +
-        `Please upgrade "prettier".`,
-    );
-  }
+  // TODO same as above
+  const prettier = prettierPath
+    ? (require(require.resolve(prettierPath, {
+        [Symbol.for('OUTSIDE_JEST_VM_RESOLVE_OPTION')]: true,
+      } as any)) as typeof import('prettier'))
+    : null;
 
   const snapshotsByFile = groupSnapshotsByFile(snapshots);
 
   for (const sourceFilePath of Object.keys(snapshotsByFile)) {
     saveSnapshotsForFile(
       snapshotsByFile[sourceFilePath],
       sourceFilePath,
-      prettier,
-      babelTraverse,
+      prettier && semver.gte(prettier.version, '1.5.0') ? prettier : null,
     );
   }
 }
 
 const saveSnapshotsForFile = (
   snapshots: Array<InlineSnapshot>,
   sourceFilePath: Config.Path,
-  prettier: Prettier,
-  babelTraverse: BabelTraverse,
+  prettier: any,
 ) => {
   const sourceFile = fs.readFileSync(sourceFilePath, 'utf8');
 
-  // Resolve project configuration.
-  // For older versions of Prettier, do not load configuration.
-  const config = prettier.resolveConfig
-    ? prettier.resolveConfig.sync(sourceFilePath, {editorconfig: true})
-    : null;
-
-  // Detect the parser for the test file.
-  // For older versions of Prettier, fallback to a simple parser detection.
-  // @ts-expect-error
-  const inferredParser: PrettierParserName = prettier.getFileInfo
-    ? prettier.getFileInfo.sync(sourceFilePath).inferredParser
-    : (config && config.parser) || simpleDetectParser(sourceFilePath);
+  // TypeScript projects may not have a babel config; make sure they can be parsed anyway.
+  const presets = [require.resolve('babel-preset-current-node-syntax')];
+  const plugins: Array<PluginItem> = [];
+  if (/\.tsx?$/.test(sourceFilePath)) {
+    plugins.push([
+      require.resolve('@babel/plugin-syntax-typescript'),
+      {isTSX: sourceFilePath.endsWith('x')},
```

Comment:
```suggestion
      {isTSX: sourceFilePath.endsWith('x')},
```

doesn't the preset figure out this on its own?

Reply:
Nope, test fails when doing that. It's probably not happening based on the `filename` passed to `parseSync` but something else in Babel that usually does it when an end user uses it :shrug: